File: Encapsulation/BankAccount.cs
public class BankAccount
{
    private double balance;

    public void Deposit(double amount)
    {
        if (amount > 0)
        {
            balance += amount;
        }
    }

    public void Withdraw(double amount)
    {
        if (amount > 0 && amount <= balance)
        {
            balance -= amount;
        }
    }

    public double GetBalance()
    {
        return balance;
    }
}

File: Encapsulation/BankAccount.ts
class BankAccount {
    private balance: number = 0;

    deposit(amount: number): void {
    if (amount > 0) {
        this.balance += amount;
    }
    }

    withdraw(amount: number): void {
    if (amount > 0 && amount <= this.balance) {
        this.balance -= amount;
    }
    }

    getBalance(): number {
    return this.balance;
    }
}
